### PR TITLE
Decimal to binary

### PR DESCRIPTION
**Pull request by ziadahanass**
Implement a solution for the challenge #76

### CODE REVIEW

File: solutions/decimal_to_binary.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        >>> decimal_to_binary(5)
+        '101'
+        >>> decimal_to_binary(0)
+        '0'
+        >>> decimal_to_binary(255)
+        '11111111'
+    """
+
+    if decimal < 0:
+        raise ValueError("Negative numbers are not supported.")
```
Comment (RandomProjects-db) [inline solutions/decimal_to_binary.py:40]: use assert as per code review requirements
better use: assert decimal >= 0, "Negative numbers are not supported."

File: solutions/decimal_to_binary.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    if decimal == 0:
```
Comment (RandomProjects-db) [inline solutions/decimal_to_binary.py:42]: Consider adding a check for decimal == 0 at the start to handle the special case immediately.

File: solutions/password_strength.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        >>> password_strength("StrongP@ssw0rd")
+        'Strong password'
+        >>> password_strength("weakpass")
+        'Weak password'
+        >>> password_strength("Short1!")
+        'Weak password'
+        >>> password_strength("NoSpecial123")
+        'Weak password'
+    """
+
```
Comment (RandomProjects-db) [inline solutions/password_strength.py:42]: It could be beneficial to return more specific error messages (e.g., "Password too short" instead of just "Weak password") to give users more useful feedback.
The function could provide more detailed feedback. For example, instead of just returning "Weak password",
return specific reasons like "Password too short", "Missing uppercase letter", etc. to help users improve their passwords.
but, it is up to you! it can still go like this.

File: solutions/tests/test_decimal_to_binary.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        """Test that decimal 255 is correctly converted to binary '11111111'."""
+        self.assertEqual(decimal_to_binary(255), "11111111")
+
+    def test_conversion_large_number(self):
+        """Test that a large decimal number is correctly converted to binary."""
+        self.assertEqual(decimal_to_binary(1024), "10000000000")
+
+    def test_negative_number(self):
+        """Test that a negative number raises an error (since negative numbers aren't supported)."""
+        with self.assertRaises(ValueError):
```
Comment (RandomProjects-db) [inline solutions/tests/test_decimal_to_binary.py:47]: Consider adding tests for extremely large decimal numbers or non-integer values (e.g., float or string) to ensure robustness.

File: solutions/tests/test_password_strength.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+
+# Debugging: Print current working directory to ensure proper location
+print("Current working directory:", os.getcwd())
+
+
+class TestPasswordStrength(unittest.TestCase):
+    """Unit tests for the password_strength function."""
+
+    def test_strong_password(self):
```
Comment (RandomProjects-db) [inline solutions/tests/test_password_strength.py:25]: It would be useful to add test cases for edge cases like an empty password, or passwords containing only digits or special characters.
You could also refactor the weak password tests to separate different reasons for weakness (e.g., no uppercase, too short).

**Review by RandomProjects-db — CHANGES_REQUESTED** [review]

**Comment by ziadahanass** [comment]
I just finished them, can you check them now?

**Review by RandomProjects-db — APPROVED** [review]

**Review by RandomProjects-db — APPROVED** [review]
